pkg/pillar/zedcloud/lookupproxy.go

```diff
@@ -121,3 +121,56 @@ func LookupProxy(status *types.DeviceNetworkStatus, ifname string,
 	log.Infof("LookupProxy: No proxy configured for port %s", ifname)
 	return nil, nil
 }
+
+// IntfLookupProxyCfg - check if the intf has proxy configured
+func IntfLookupProxyCfg(status *types.DeviceNetworkStatus, ifname, downloadURL string) string {
+	var mayChange bool
+	// if proxy is not on the intf, then don't change anything
+	// if download URL has "http://" or "https://" then no change here regardless of proxy
+	// if there is proxy on this intf, treeat empty url scheme as for https,
+	// treat passed in download url schem docker://" and maybe sftp://just as https
```

Comment:
s/schem/sheme/

More importantly, can we change a docker:// URL to https:// ? Or is that not what the comment is saying?

Reply:
right, the comment is saying will try to replace, if the intf is a proxy with http/https.